copier/main.py

```diff
@@ -449,6 +449,10 @@ def jinja_env(self) -> SandboxedEnvironment:
         env.filters["to_json"] = partial(
             env.filters["to_json"], default=pydantic_encoder
         )
+        # Add a helper function to join filesystem paths using the path
+        # separator `/` as Jinja uses the POSIX and not OS-specific path
+        # separator for includes/imports.
+        env.globals["posixpath"] = lambda *p: "/".join(p)
```

Comment:
WDYT about calling it `posixjoin` and using it as filter instead?

    "Base" | posixjoin("next", "next")| posixjoin ("woohoo more!")

Reply:
Or even better: `pathjoin`, with a kwarg `mode` that accepts "native" , "windows" or "posix" (default).

Reply:
Would you prefer a filter over a global function because the global function pollutes the context? E.g. a template creator would not be able to name a question `pathjoin` (after the renaminh you suggested)? Or is there a particular scenario you have in mind where chaining `pathjoin` filters comes in handy?

I'm not opposed to the filter approach, just trying to understand your thoughts.

Reply:
I went ahead and made the change as you suggested. I'd still be curious about your thoughts.

Reply:
Mhh, CI on Windows is failing. I'll take a look tomorrow.

Reply:
@yajo I'm afraid using a filter in a folder of file name won't work because the [`|` character is not allowed there on Windows](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file?redirectedfrom=MSDN#naming-conventions). That's why the CI job in Windows is failing and I've manually verified it on a Windows machine as well. I suggest to register `pathjoin()` as a function (similar to before):

```jinja
{% include pathjoin('includes', 'name-slug.jinja') %}.py
```

I'd keep the optional kwarg `mode` of course. WDYT?

Reply:
It's not critical for this particular feature, but I wonder whether maintaining Windows compatibility is worth the sacrifice of not being able to use Jinja filters in folder and file names in general. :thinking: My experience with developing software on Windows is awful. And since there is WSL, Windows users can escape into a sane development environment.

Reply:
Yes, maintaining windows is terrible. However it's more or less done already, so let's stick with the function solution. The filter looked cooler to me, but given this technical limitation, I think it's the best.

Reply:
Alright, I've reverted `pathjoin()` to a function.